### ISSUE TITLE
Feature: Share Button

### ISSUE THREAD

**Issue by stardust66** [ChristopherChudzicki/math3d#68]
When clicked, shows a modal dialog or something else displaying the url associated with the graph. Something in the form of `http://<domain>/<username>/<graph_id>`.

**Comment by ChristopherChudzicki** [comment]
@stardust66 Great job on your [introduce-flask](https://github.com/stardust66/math3d/tree/introduce-flask) branch—you've added a login and given users the ability to store graphs with short URLs.

One suggestion from my brother is that users should not be required to log in in order to generate a short, shareable URL. But login is still useful so we can display a collection of graphs the user has saved for their convenience. 

I suggest two separate features:

1. **Share button**: Maps short, random, unique id to graph settings. Also generate (via a modal or popup) a url of the form `http://<domain>/<graph_id>` that user can share.
2. **Save Button**: Maps short, random, unique id to graph settings. Also adds graph to user's list of saved graphs. If a logged-in user loads one of their previous graphs and _then_ clicks save, they should be given a choice to save as new or save over existing graph. Probably save as new should be default.

The [Save] and [Share] buttons should generate completely separate ids. (If user saves a graph, then shares, then edits existing graph, the shared graph should _not_ be affected). This might lead to duplicate saved settings in our files, but who cares. 

In terms of implementation, [Share] seems to me (1) higher priority and (2) easier. In fact, @stardust66's  [introduce-flask](https://github.com/stardust66/math3d/tree/introduce-flask) branch probably has 95% of what is needed, with some extra stuff that can be removed (for now).

**Comment by stardust66** [comment]
On my new branch `flask-production`, I deleted the login and registration related functionalities and made Graph objects not require a username. I'm using a md5 hash for storing and retrieving graphs. Visiting `/graph/<graph_hash>` will make the server retrieve the corresponding graph. This isn't finished but the changes are live on [math3dstaging.herokuapp.com](http://math3dstaging.herokuapp.com).

I looked at Desmos, and the way they do this is, regardless of whether the graph has been saved previously, create a new unique url. Even when saving an edited graph, a new url is generated. The old url still points to the old, unchanged graph. I don't know if this is going to take up a lot of database space but it's probably viable for now because people probably aren't going to create so many graphs as to exceed heroku's limit, and we can always move to something else.
